### PR TITLE
Implement the FNV hash library for the probabilistic sampler

### PR DESCRIPTION
**Pull request by moonbox3**
**Description:** 
This change moves to use the FNV hash library as is used in other modules, instead of using a custom implementation of a murmur3 hash. Cleaned up some of the existing code as it didn't need explicit conditional branches.

**Link to tracking Issue:** #16456

**Testing:** some unit tests were updated to account for the change in how the hash is generated and what that means for events that were sampled. All unit tests are passing.

**Documentation:** No documentation update is required.

**Changelog:** Added a changelog based on the enhancement.

### CODE REVIEW

**Review by MovieStoreGuy — COMMENTED** [review]
I am happy with the changes however, could I ask you to make a changelog as an enhancement since we are changing the hashing algorithm?

**Comment by moonbox3** [comment]
> I am happy with the changes however, could I ask you to make a changelog as an enhancement since we are changing the hashing algorithm?

@MovieStoreGuy Thank you for reviewing. Yes, of course, will work on this shortly.

**Comment by moonbox3** [comment]
@MovieStoreGuy the changelog is included.

**Review by MovieStoreGuy — APPROVED** [review]

File: processor/probabilisticsamplerprocessor/logsprocessor_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func TestLogsSampling(t *testing.T) {
 				SamplingPercentage: 50,
 				AttributeSource:    traceIDAttributeSource,
 			},
-			received: 52,
+			received: 45, // 52
```
Comment (dmitryax) [inline processor/probabilisticsamplerprocessor/logsprocessor_test.go:101]: What does the comment mean?

Comment (moonbox3) [reply processor/probabilisticsamplerprocessor/logsprocessor_test.go]: The comment is not needed. Good catch, thank you. I removed it.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**